R/dp_init.R

```diff
@@ -77,6 +77,14 @@ dp_init <- function(project_path = fs::path_wd(),
   commit_description <- "dp init"
   wd0 <- fs::path_wd()
 
+  if(fs::dir_exists(project_path)){
```

Comment:
Addition of this leads to confusion on what the desired behavior should be based on line 88:
- line 80 says -> stop if dir doesn't exist
- line 88 says -> create if dir doesn't exist
One should be chosen and the other deleted.  I suggest we remove line 80 as the issue it is try to address is addressed by line 84 `is_valid_dp_repository(project_path)`

Reply:
note `is_valid_dp_repository(project_path)` checks all `all` elements of the repository ("git" & "dp" & "renv" & "branch"). We may want an `or` logic so that if any of those elements are passed it returns TRUE and hence stops

Reply:
@amashadihossein 

I made the changes to address the issues. `is_valid_dp_repository` has been revised to check for all ("git" & "dp" & "renv" & "branch") versus any checks. Please let me know if this seems alright or we may want to seek a way around it. Thanks.

Reply:
The error message you have on line 81 of dp_init.R specifically references a dp repository. So if the user is in git or renv repository (but not a dp repository) they will get the error message "You are already inside a dp repository". Recommend changing the error message to something more specific to which of the checked elements already exist. There's a verbose option on `is_valid_dp_repository` that could probably be used here.